Code/GraphMol/SmilesParse/smarts.yy

```diff
@@ -338,28 +339,50 @@ hydrogen_atom:	ATOM_OPEN_TOKEN H_TOKEN ATOM_CLOSE_TOKEN
 atom_expr: atom_expr AND_TOKEN atom_expr {
   $1->expandQuery($3->getQuery()->copy(),Queries::COMPOSITE_AND,true);
   if($1->getChiralTag()==Atom::CHI_UNSPECIFIED) $1->setChiralTag($3->getChiralTag());
+  SmilesParseOps::ClearAtomChemicalProps($1);
```

Comment:
Do you need to clear the chemical props when anding to a different prop?  I would expect that this only needs to be cleared when an or is involved.